### ISSUE TITLE
epsilon in `DiceLoss` seems a bit high, which results in unintuitive results

### ISSUE THREAD

**Issue by MuadDev** [qubvel/segmentation_models.pytorch#497]
The DiceLoss declared here has epsilon set to 1:
https://github.com/qubvel/segmentation_models.pytorch/blob/master/segmentation_models_pytorch/utils/losses.py

Running this piece of code results in an unexpected loss value
```python
import torch
from segmentation_models_pytorch.utils.losses import DiceLoss

dice_loss = DiceLoss()

batch_size = 1
num_classes = 1
image_size = 10
zeros = torch.zeros(size=(batch_size, num_classes, image_size, image_size), dtype=torch.float64)
ones = torch.ones(size=(batch_size, num_classes, image_size, image_size), dtype=torch.float64)
print(dice_loss(ones, zeros))  # Prints 0.9901, would've expected 1 instead
```

Setting the epsilon to a sufficiently low value solves this error:
```python
import torch
from segmentation_models_pytorch.utils.losses import DiceLoss

dice_loss = DiceLoss(
    eps=1e-7
    )

batch_size = 1
num_classes = 1
image_size = 10
zeros = torch.zeros(size=(batch_size, num_classes, image_size, image_size), dtype=torch.float64)
ones = torch.ones(size=(batch_size, num_classes, image_size, image_size), dtype=torch.float64)
print(dice_loss(ones, zeros))  # Prints 1
```

Perhaps it would be better to set the default epsilon lower to prevent this?

**Comment by JulienMaille** [comment]
Discussed here #456 but I'm not sure what's the bottom line
